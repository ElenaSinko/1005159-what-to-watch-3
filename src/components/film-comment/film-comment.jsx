import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

export const FilmComment = ({comment}) => {
  return <div className="review">
    <blockquote className="review__quote">
      <p className="review__text">{comment.comment}</p>

      <footer className="review__details">
        <cite className="review__author">{comment.user.name}</cite>
        <time className="review__date" dateTime="2015-11-18">{moment(comment.date).format(`MMMM DD, YYYY`)}</time>
      </footer>
    </blockquote>

    <div className="review__rating">{comment.rating}</div>
  </div>;
};

FilmComment.propTypes = {
  comment: PropTypes.object,
};

