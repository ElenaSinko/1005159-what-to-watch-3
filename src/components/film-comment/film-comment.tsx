import * as React from "react";
import * as moment from 'moment';
import {Comment} from "../../types";

interface Props {
  comment: Comment;
}

export const FilmComment: React.FunctionComponent<Props> = (props: Props) => {
  const {comment} = props;
  return <div className="review">
    <blockquote className="review__quote">
      <p className="review__text">{comment.comment}</p>

      <footer className="review__details">
        <cite className="review__author">{comment.user.name}</cite>
        <time className="review__date" dateTime="2015-11-18">{moment(comment.date).format(`MMMM DD, YYYY`)}</time>
      </footer>
    </blockquote>

    <div className="review__rating">{comment.rating}</div>
  </div>;
};
