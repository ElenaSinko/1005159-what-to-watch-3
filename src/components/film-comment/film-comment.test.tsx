import * as React from "react";
import * as renderer from "react-test-renderer";
import {FilmComment} from "./film-comment";

const userComment = {
  id: 1,
  user: {id: 19, name: `Christina`},
  rating: 9.5,
  comment: `This movie really touched my heart.`,
  date: `2020-00-28T11:05:16.526Z`,
};

it(`Render FilmComment`, () => {
  const tree = renderer
    .create(<FilmComment
      comment={userComment}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
