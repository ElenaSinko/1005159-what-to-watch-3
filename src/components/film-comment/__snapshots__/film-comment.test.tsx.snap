// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FilmComment 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      This movie really touched my heart.
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Christina
      </cite>
      <time
        className="review__date"
        dateTime="2015-11-18"
      >
        Invalid date
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    9.5
  </div>
</div>
`;
